runner/runners/invoke.go

```diff
@@ -283,98 +273,105 @@ func (inv *Invoker) run(cmd *runner.Command, id runner.RunID, abortCh chan struc
+			inv.stat.Counter(stats.WorkerUploads).Inc(1)
+			defer func() {
+				os.RemoveAll(tmp.Dir)
+				uploadTimer.Stop()
+			}()
+			outPath := stdout.AsFile()
+			errPath := stderr.AsFile()
+			logPath := stdlog.AsFile()
+			stdoutName := "STDOUT"
+			stderrName := "STDERR"
+			stdlogName := "STDLOG"
 
-		writer.Close()
-		reader.Close()
-		if writer, err = os.Create(filepath.Join(tmp.Dir, stderrName)); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stderr: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if reader, err = os.Open(errPath); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stderr: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if _, err := io.Copy(writer, reader); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stderr: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		}
+			if err = copyLogFile(tmp.Dir, stdoutName, outPath); err != nil {
+				return runner.FailedStatus(id, err,
+					tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
+			}
+			if err = copyLogFile(tmp.Dir, stderrName, errPath); err != nil {
+				return runner.FailedStatus(id, err,
+					tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
+			}
+			if err = copyLogFile(tmp.Dir, stdlogName, logPath); err != nil {
+				return runner.FailedStatus(id, err,
+					tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
+			}
 
-		writer.Close()
-		reader.Close()
-		if writer, err = os.Create(filepath.Join(tmp.Dir, stdlogName)); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdlog: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if reader, err = os.Open(logPath); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdlog: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if _, err := io.Copy(writer, reader); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdlog: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		}
+			ingestCh := make(chan interface{})
+			go func() {
+				snapshotID, err := inv.filerMap[runType].Filer.Ingest(tmp.Dir)
+				if err != nil {
+					ingestCh <- err
+				} else {
+					ingestCh <- snapshotID
+				}
+			}()
 
-		ingestCh := make(chan interface{})
-		go func() {
-			snapshotID, err := inv.filerMap[runType].Filer.Ingest(tmp.Dir)
-			if err != nil {
-				ingestCh <- err
-			} else {
-				ingestCh <- snapshotID
+			// Meaningful to support Abort after execer has completed?
+			var snapshotID string
+			select {
+			case <-abortCh:
+				return runner.AbortStatus(id, tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
```

Comment:
just curious.... this doesn't really abort the ingest does it?

Reply:
It lets the worker go and do something else, but the actual ingestion keeps going (same as existing behavior)